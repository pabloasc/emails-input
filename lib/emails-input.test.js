const EmailsInput = require('emails-input');
const { mainModule } = require('process');

describe('EmailsInput tests', () => {
    test('randomEmail expect random email to be generated', () => {
        const container = document.createElement('div');
        const emailInputComponent = EmailsInput(container, {});
        expect(emailInputComponent.randomEmail()).toMatch(/@mail.ru/);
    });

    test('emailIsValid expect to validate emails', () => {
        const container = document.createElement('div');
        const emailInputComponent = EmailsInput(container, {});
        expect(emailInputComponent.emailIsValid('tim@mail.ru')).toBe(true);
        expect(emailInputComponent.emailIsValid('timmail.ru')).toBe(false);
        expect(emailInputComponent.emailIsValid('tim@mailru')).toBe(false);
    });


    it('Email block to match snapshot', () => {
        const container = document.createElement('div');
        const emailInputComponent = EmailsInput(container, {});
      
        expect(emailInputComponent.emailBlock('email@mail.ru')).toMatchSnapshot();
    });
});