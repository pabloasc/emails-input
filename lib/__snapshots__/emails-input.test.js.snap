// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailsInput tests will fail every time 1`] = `
<span
  class="email-block"
>
  <span
    id="email-text-container"
  >
    email@mail.ru
  </span>
  <span
    id="email-remove"
  >
    <svg
      height="8"
      viewBox="0 0 8 8"
      width="8"
    >
      <path
        d="M8 0.8L7.2 0L4 3.2L0.8 0L0 0.8L3.2 4L0 7.2L0.8 8L4 4.8L7.2 8L8 7.2L4.8 4L8 0.8Z"
        fill="#050038"
      />
    </svg>
  </span>
</span>
`;
